src/eHealth.locations/services/selected-location-factory.js

```diff
@@ -246,8 +246,19 @@ angular.module('eHealth.locations.services')
             }
           });
           return angular.copy(adminDivisions);
+        },
+        clear: function () {
+          levels.forEach(function (level) { delete level.selected });
         }
       };
+
+      Object.defineProperty(location, 'innermost', {
```

Comment:
This is a very different style compared to how the other methods of `location` ore defined. It is also more verbose. Is there any advantage in using `Object.defineProperty`.